skimage/_shared/kahan.pyx

```diff
@@ -0,0 +1,86 @@
+# -*- python -*-
+#cython: cdivision=True
+#cython: boundscheck=False
+#cython: nonecheck=False
+#cython: wraparound=False
+
+import numpy as np
+import cython
+
+
+def fsum_vector(cython.floating[:] v, int calc_cumsum=0):
+    """Apply Kahan summation to a vector.
+
+    Parameters
+    ----------
+    v : ndarray of double
+        Array to sum.
+    calc_cumsum : int
+        Whether or not to calculate the cumulative sum as well.
+
+    References
+    ----------
+    .. [1] https://en.wikipedia.org/wiki/Kahan_summation_algorithm
+    """
+    cdef:
+        double s = 0
+        double y, t
+        double c = 0
+        Py_ssize_t i
+        cython.floating[:] cumsum
+
+    if calc_cumsum:
+        cumsum = np.zeros_like(v)
+
+    for i in range(v.size):
+        y = v[i] - c
```

Comment:
I know that the symbols (y, v, c, ..) are the ones used in the wikipedia page, but one-letter names are really hard to understand. Change to more meaningful names such as "compensation" or "running_compensation", "compensated_value", "compensated_sum" etc. ?